### ISSUE TITLE
terraform destroy fails to destroy DynamoDB tables.

### ISSUE THREAD

**Issue by gyoju** [nasa/cumulus#1388]
```
Error: Instance cannot be destroyed

  on .terraform/modules/data_persistence/tf-modules/data-persistence/dynamo.tf line 18:
  18: resource "aws_dynamodb_table" "access_tokens_table" {

Resource module.data_persistence.aws_dynamodb_table.access_tokens_table has
lifecycle.prevent_destroy set, but the plan calls for this resource to be
destroyed. To avoid this error and continue with the plan, either disable
lifecycle.prevent_destroy or reduce the scope of the plan using the -target
flag.
```

**Comment by nemreid** [comment]
Hi @gyoju, unfortunately this is working as intended. In order for us to protect database resources in a production environment we set `lifecycle.prevent_destroy = true` on each DynamoDB table.

Unfortunately we cannot make this configurable as this is a limitation with Terraform. I refer you to hashicorp/terraform#3116.

We decided it was preferable to impose the slightly onerous requirement of deleting the 10 tables by hand rather than have no protections against deletion in a production environment. Sorry for the inconvenience! We feel it too in our development work, and I can assure you that if Terraform starts to allow a configurable value for this setting we will incorporate it into our project as well.
